src/core/org/apache/jmeter/threads/JMeterThread.java

```diff
@@ -238,9 +248,10 @@ private void startScheduler() {
     public void setThreadName(String threadName) {
         this.threadName = threadName;
     }
-
+    public static final ThreadLocal <Boolean>threadLocal4SameUser = new ThreadLocal<Boolean>() ;
```

Comment:
Request was to make this field not Thread Local but a private JMeter Variable similar to the ones prefixed by __jmeter.....